### PR TITLE
Add xnat and container service roles

### PR DESCRIPTION
**Pull request by drmatthews**
Changes:

- Add `xnat` role
- Add `xnat_container_service` role

### CODE REVIEW

File: playbooks/group_vars/web.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+firewalld_internal_zone_open_services:
+  - http
+  - https
+  - ssh
+firewalld_public_zone_open_services:
+  - http
+  - https
+firewalld_work_zone_open_services:
+  - http
+  - https
```
Comment (p-j-smith) [inline playbooks/group_vars/web.yml:78]: do these need to be updated in the [xnat collection](https://github.com/UCL-MIRSG/ansible-collection-xnat/blob/38d48cea16c768d73f803de2019c0c80cefe733b/playbooks/group_vars/web.yml#L68-L78) for it to work correctly?

File: playbooks/group_vars/db.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  use_pk8: false
+  csr_filename: "{{ postgresql.base_directory }}/certs/server.csr"
+  csr_common_name: "{{ xnat_db.host }}"
+  certificate_filename: "{{ postgresql.base_directory }}/certs/server.crt"
+  provider: "selfsigned"
+  cache_filename: "{{ database_server_certificate_cache_filename }}" # where to store the server certificate in cache
+
+firewalld_rich_rules:
+  - zone: "internal"
+    rule: "family=ipv4 source address={{ xnat_web_server.subnet | default(xnat_web_server.ip + '/32') }} port protocol=tcp port={{ xnat_db.port }} accept"
```
Comment (p-j-smith) [inline playbooks/group_vars/db.yml:42]: this variable name doesn't have the `firewalld_` prefix in the[ xnat collection](https://github.com/UCL-MIRSG/ansible-collection-xnat/blob/38d48cea16c768d73f803de2019c0c80cefe733b/playbooks/group_vars/db.yml#L40-L42), but I think it's correct here?

**Review by p-j-smith — CHANGES_REQUESTED** [review]
looks good! It'll be great to have one repo for these roles / playbooks

For the xnat and container service roles, was anything else changed from the `main` branch except the variable names I've commented on (which I think are correct here)?

Couple of things that would be good to add (happy for them to be a separate PR if that's easier though):
- update the renovate config to include the packages in the [xnat collection config](https://github.com/UCL-MIRSG/ansible-collection-xnat/blob/main/.renovaterc.json)
- add the [xnat collection architecture notes](https://github.com/UCL-MIRSG/ansible-collection-xnat/blob/main/architecture_notes.md) (possible in the `playbooks/` directory?)

File: playbooks/group_vars/web.yml (hunk 1 above)
Comment (drmatthews) [reply playbooks/group_vars/web.yml:78]: They do - can't believe I missed those. I've opened [another PR](https://github.com/UCL-MIRSG/ansible-collection-xnat/pull/20).

File: playbooks/group_vars/db.yml (hunk 2 above)
Comment (drmatthews) [reply playbooks/group_vars/db.yml:42]: See above about [this PR](https://github.com/UCL-MIRSG/ansible-collection-xnat/pull/20).

File: playbooks/group_vars/web.yml (hunk 1 above)
Comment (p-j-smith) [reply playbooks/group_vars/web.yml:78]: thanks!

**Comment by drmatthews** [comment]
I didn't change anything else in the roles - moved them over as-is. I'd intended to update the `firewalld` variable names in the `xnat` collection first but I managed to miss the ones you pointed out.

**Review by p-j-smith — APPROVED** [review]
nice work!

**Pull request merged**